x-pack/legacy/plugins/reporting/server/lib/validate/validate_config.ts

```diff
@@ -5,15 +5,19 @@
  */
 
 import crypto from 'crypto';
-import { Logger } from '../../../types';
+import { ServerFacade, Logger } from '../../../types';
+
+export function validateConfig(serverFacade: ServerFacade, logger: Logger) {
+  const config = serverFacade.config();
 
-export function validateConfig(config: any, logger: Logger) {
   const encryptionKey = config.get('xpack.reporting.encryptionKey');
   if (encryptionKey == null) {
     logger.warning(
       `Generating a random key for xpack.reporting.encryptionKey. To prevent pending reports from failing on restart, please set ` +
         `xpack.reporting.encryptionKey in kibana.yml`
     );
-    config.set('xpack.reporting.encryptionKey', crypto.randomBytes(16).toString('hex'));
+
+    // @ts-ignore: No set() method on KibanaConfig, just get() and has()
+    config.set('xpack.reporting.encryptionKey', crypto.randomBytes(16).toString('hex')); // update config in memory to contain a usable encryption key
```

Comment:
Hm, this part in particular might need a followup, if `.set` is not in the type because it is going away...